internal/management/controller/instance_controller.go

```diff
@@ -313,10 +314,14 @@ func (r *InstanceReconciler) restartPrimaryInplaceIfRequested(
 		); err != nil {
 			return true, err
 		}
-		oldCluster := cluster.DeepCopy()
-		cluster.Status.Phase = apiv1.PhaseHealthy
-		cluster.Status.PhaseReason = "Primary instance restarted in-place"
-		return true, r.client.Status().Patch(ctx, cluster, client.MergeFrom(oldCluster))
+
+		return true, clusterstatus.RegisterPhase(
+			ctx,
+			r.client,
+			cluster,
+			apiv1.PhaseHealthy,
+			"Primary instance restarted in-place",
+		)
```

Comment:
I have been wondering why this piece of code doesn't generate a conflict.
I discovered that both the operator and the instance manager identify themselves as "manager" in the ManagedFields. 
If we ever change this behavior in the future, the code here will trigger a conflict.